tests/conftest.py

```diff
@@ -44,6 +43,113 @@ def teardown():
     request.addfinalizer(teardown)
 
 
+def _add_framework(request, app, slug, status, framework, framework_agreement_details=None):
+    if framework_agreement_details:
+        framework_agreement_details = {'frameworkAgreementVersion': 'v1.0'}
+
+    with app.app_context():
+        framework = Framework(
+            slug=slug,
+            name=slug,
+            framework=framework,
+            framework_agreement_details=framework_agreement_details,
+            status=status,
+            clarification_questions_open=True if status == 'open' else False
+        )
+        db.session.add(framework)
+        db.session.commit()
+
+        def teardown():
+            with app.app_context():
+                framework = Framework.query.filter(Framework.slug == slug).first()
+                Framework.query.filter(Framework.id == framework.id).delete()
+                db.session.commit()
+
+    request.addfinalizer(teardown)
+
+
+def _base_framework(
+        request, app, slug, status, framework, framework_agreement_details=None
+):
+    with app.app_context():
+        if Framework.query.filter(Framework.slug == slug).first():
+            _update_framework_status(request, app, slug=slug, status=status)
+        else:
+            _add_framework(
+                request,
+                app,
+                slug=slug,
+                status=status,
+                framework=framework,
+                framework_agreement_details=framework_agreement_details
+            )
+
+
+def _g8_framework(request, app, status):
+    _base_framework(
+        request, app, slug='g-cloud-8', status=status, framework='g-cloud', framework_agreement_details=True)
+
+
+def _g7_framework(request, app, status):
+    _base_framework(
+        request, app, slug='g-cloud-7', status=status, framework='g-cloud', framework_agreement_details=False)
+
+
+def _g6_framework(request, app, status):
+    _base_framework(
+        request, app, slug='g-cloud-6', status=status, framework='g-cloud', framework_agreement_details=False)
+
+
+def _dos_framework(request, app, status):
+    _base_framework(
+        request,
+        app,
+        slug='digital-outcomes-and-specialists',
+        status=status,
+        framework='dos',
+        framework_agreement_details=False
+    )
+
+
+# G8
+
+
+@pytest.fixture()
```

Comment:
Tried it and tests started breaking.  

Didn't really look into it.

Feel free to refactor if you like!